core/cap-0034.md

```diff
@@ -0,0 +1,352 @@
+events that could lead to a smart contract executing inconsistently because of
+the problem that this CAP intends to address might be very rare, they might
+also be of a kind of which even a single event could significantly damage trust.
+A "reliable" network must try to avoid _any_ behavior which could appear
+sufficiently unpredictable to a client that they would be unlikely to consider
+it and could obtain inconsistent results when it happened.
+
+- The Stellar Network should run at scale and at low cost to all users.
+Keeping transactions which are guaranteed to fail later in
+externalized transaction sets is wasteful.
+
+It also aligns with the following Stellar Network Value:
+
+- The protocol should bias toward simplicity.  This CAP simplifies the
+ballot protocol by removing the "combination" of `closeTime`s and simply
+selecting the one associated with the selected transaction set.
+
+## Abstract
+
+We change the (deterministic) mechanism by which the ballot protocol computes
+the `StellarValue` to be externalized from producing a value comprising the
+best transaction set (selected by a deterministic heuristic which favors larger
+sets) of any candidate, the maximum `closeTime` of any candidate, and a set of
+"maximal" ledger `upgrades` from all candidates to one which takes the
+`closeTime` from the same candidate `StellarValue` as the best transaction set
+(the heuristic which decides which transaction set is best does not change;
+nor does the method of generating `upgrades`, which remains a combination).
+
+In particular, this affects how soon the core notices when transactions expire.
+
+## Specification
+
+There are no changes to any XDR in this CAP.  The treatment of the `closeTime`
+in the `StellarValue` XDR in some code paths changes, as does the use of
+`SIGNED` `StellarValue`s in some places where `BASIC` ones are currently used,
+but the XDR itself does not change.
+
+To date, the ballot protocol has generated a value to externalize by combining
+all candidates in a deterministic way:  the transaction set comes from a
+"best-transaction-set" heuristic (generally, bigger transaction sets are
+"better"), the closetime is the maximum of all of the candidates' closetimes,
+and the set of ledger upgrades is one ledger upgrade type per type represented
+among all candidates, where multiple upgrades of a given type are combined into
+a single upgrade of that type by a type-specific algorithm (in general --
+though, as it happens, for current upgrade types, all of the algorithms are
+"maximum value", like the closetime combination algorithm).
+
+This CAP proposes to change the externalized-value generation to remove the
+combining of closetimes and simply select the one from the same nominated
+value as the best transaction set.  (It proposes to preserve the selection
+heuristic for "best transaction set" unchanged.)
+
+Because the closetime and transaction set in the externalized value would
```

Comment:
Again, there isn't really a need to justify why you are doing things here. Just tell the reader in simple terms what it is. The goal is to make this section as precise and concise as possible so that it is easy to turn it directly into an implementation. Justifications you can move to the design rationale section.

It is probably worth revisiting this entire section with this goal in mind.

Reply:
Okay, I've tried to trim the specification down to just specifying.  The justifications were, I think, just redundant with the ones already in the "Design Rationale".